flink-runtime/src/main/java/org/apache/flink/runtime/security/FlinkSecurityManager.java

```diff
@@ -217,4 +200,19 @@ void unmonitorUserSystemExit() {
     boolean userSystemExitMonitored() {
         return Boolean.TRUE.equals(monitorUserSystemExit.get());
     }
+
+    /**
+     * Use this method to circumvent the configured {@link FlinkSecurityManager} behavior, ensuring
+     * that the current JVM process will always stop via System.exit() or
+     * Runtime.getRuntime().halt().
+     */
+    public static void forceProcessExit(int exitCode) {
+        // Unset ourselves to allow exiting in any case.
+        System.setSecurityManager(null);
+        if (flinkSecurityManager != null && flinkSecurityManager.haltOnSystemExit) {
+            Runtime.getRuntime().halt(exitCode);
+        } else {
+            System.exit(exitCode);
+        }
```

Comment:
we probably don't have to bother, but there is a small window here where user-code could also cause failures or any code can do other stuff that security managers would usually enforce.

I'm wondering whether add a switch to the FlinkSM to specifically disable the system exit rules.
We then wouldn't set the SM to null, nor would have to evaluate haltOnSystemExit; we just call the switch and the System.exit.

Reply:
I would consider this a follow up improvement, if somebody really needs this. If a user wants to use their own security manager and protect against this, they can forbid access to replacing the security manager, using the "setSecurityManager" permission.